zuul-core/src/main/java/com/netflix/zuul/netty/connectionpool/DefaultClientChannelManager.java

```diff
@@ -225,6 +238,8 @@ public boolean release(final PooledConnection conn) {
         discoveryResult.incrementNumRequests();
 
         if (shuttingDown) {
+            LOG.debug("[{}] closing connection released during shutdown", conn.getChannel().id());
+            conn.close();
```

Comment:
The pool loses track of a connection as soon as it is borrowed. Since we're shutting down it seemed simplest to just immediately close in this case vs returning to to the pool and trying to close there